rfcs/rfc0018.md

```diff
@@ -0,0 +1,69 @@
+# A feature to automatically "yield true" at the end of a file
+
+## Preamble
+
+    Author:  Curtis "Ovid" Poe <curtis.poe@gmail.com>
+    Sponsor:
+    ID:      0018
+    Status:  Draft
+
+## Abstract
+
+This RFC proposes a `yield_true` feature. When used, the current Perl _file_
+containing `use feature 'yield_true'l` will automatically return a true value
+after successful compilation, eliminating the neeed for a "1" (or other true
+value) at the end of the file.
+
+## Motivation
+
+Eliminate the need for a true value at the end of a Perl file.
+
+## Rationale
+
+There's no need to have a true value be hard-coded in our files that we
+`use`. Further, newer programmers can get confused because sometimes code
+_doesn't_ end with a true value but nonetheless compiles just fine because
+_something_ in the code returned a true value and the code compiles as a
+side-effect.
+
+
+## Specification
+
+    use feature 'yield_true';
+
+Code using the above does not need to return a magic true value when compiled.
```

Comment:
Something to think about is that while *most* modules just put "1" or some other dummy true value, there are some modules that make use of the return value, because it becomes the value returned by a `require`. In particular, it is not uncommon to yield the package name itself, when that is an object class, so you can do things like

```perl
my $obj = (require Some::Object::Class)->new;
```

I don't think we want to complicate this feature idea by suggesting we'd allow customisation of the true value being returned, but perhaps if we're returning some value anyway, the string form of the package name is just as good as any, and makes this use-case still work. (And anyone who writes `package 0` deserves what they get :P)